test/test.ts

```diff
@@ -1361,6 +1361,19 @@ describe('gcs-resumable-upload', () => {
       it('should return true', () => {
         assert.strictEqual(up.onResponse(RESP), true);
       });
+
+      it('should handle a custom status code when passed a retry function', () => {
+        const RESP = {status: 1000};
+        const customHandlerFunction = (err: ApiError) => {
+          if (err.code === 1000) {
+            return true;
+          }
+          return false;
+        };
+        up.retryableErrorFn = customHandlerFunction;
+
+        assert.strictEqual(up.onResponse(RESP), false);
```

Comment:
Not sure why this should be false. The function we're passing in says to return `true` for 1000?

Reply:
This is not testing the return value of the `retryableErrorFn` but rather the return value of the `onResponse` function. `onResponse` returns false when it attempts a retry, which it should in this case when the `retryableErrorFn` tells it that this is a retryable status code.